tests/foreman/ui_airgun/test_hostcollection.py

```diff
@@ -635,3 +702,89 @@ def test_negative_hosts_limit(session, module_org, module_loc):
         assert "cannot have more than 1 host(s) associated with host collection '{}'".format(
             hc_name
         ) in str(context.value)
+
+
+@tier3
+@upgrade
+def test_positive_install_module_stream(
+        session, module_org, vm_content_hosts_module_stream, vm_host_collection_module_stream):
+    """Install a module-stream to hosts inside host collection remotely
+
+    :id: e5d882e0-3520-4cb6-8629-ef4c18692868
+
+    :Steps:
+        1. Running dnf upload profile to sync module streams from hosts to Satellite
+        2. Navigating to host_collection
+        3. Installing the module stream duck
+        4. Verifying that remote job get passed
+        5. Verifying that package get installed
+
+    :expectedresults: Module-Stream was successfully installed on all the hosts
+        in host collection
+
+    :CaseLevel: System
+    """
+    with session:
+        _run_remote_command_on_content_hosts(
+            'dnf -y upload-profile',
+            vm_content_hosts_module_stream
+        )
+        session.organization.select(org_name=module_org.name)
+        result = session.hostcollection.manage_module_streams(
+            vm_host_collection_module_stream.name,
+            action_type="Install",
+            module_name=FAKE_3_CUSTOM_PACKAGE_NAME,
+            stream_version="0"
+        )
+
+        assert result['overview']['job_status'] == 'Success'
+        assert result['overview']['job_status_progress'] == '100%'
+        assert int(result['overview']['total_hosts']) == 2
+        assert _is_package_installed(vm_content_hosts_module_stream, FAKE_3_CUSTOM_PACKAGE)
+
+
+@tier3
+@upgrade
+def test_positive_install_modular_errata(
+        session, module_org, vm_content_hosts_module_stream, vm_host_collection_module_stream):
+    """Install Modular Errata generated from module streams.
+
+    :id: 8d6fb447-af86-4084-a147-7910f0cecdef
+
+    :Steps:
+        1. Generated errata by installing and downgrading the Module Stream Kangaroo
+        2. Running dnf upload-profile
+        3. Installed the modular errata by 'remote execution'
+        4. Verifying that latest package get installed
+
+    :expectedresults: Modular Errata should get installed on all hosts in host
+        collection.
+
+    :CaseLevel: System
+    """
+    with session:
+        stream = "0"
+        version = "20180704111719"
```

Comment:
what is that magic number?

Reply:
That's a version of module stream kangaroo (hence I enclosed by a variable name). Not a magic number :)

Reply:
That is too specific and is too... yeah specific. Usually, it is really bad idea to use something like this in test automation as it quickly become out of date. Do we have any alternatives to avoid that number?

Reply:
quietly I dont have any alternative for that. It is actually an alternative of   https://github.com/SatelliteQE/robottelo/pull/6906#discussion_r283326090 comment. Would you prefer that?

Reply:
By alternative I meant something like `3.0.0.1` is better than `340958349058903458`. Or we don't have versions readable by humans there? :)

Reply:
@oshtaier I don't think so or not which I aware about. 

```
[root@kvm-02-guest19 ~]# dnf module info kangaroo 
Updating Subscription Management repositories.
Copr repo for swid owned by jpazdzio                                                                     55 kB/s | 3.0 kB     00:00    
swid_zoo_repo                                                                                            51 kB/s | 3.3 kB     00:00    
Name             : kangaroo
Stream           : 0 [d][e][a]
Version          : 20180704111719
Context          : deadbeef
Profiles         : default [d] [i]
Default profiles : default
Repo             : SWID_Testing_SWID_Product_swid_zoo_repo
Summary          : Kangaroo 0.2 module
Description      : A module for the kangaroo 0.2 package
Artifacts        : kangaroo-0:0.2-1.noarch

Name             : kangaroo
Stream           : 0 [d][e][a]
Version          : 20180730223407
Context          : deadbeef
Profiles         : default [d] [i]
Default profiles : default
Repo             : SWID_Testing_SWID_Product_swid_zoo_repo
Summary          : Kangaroo 0.3 module
Description      : A module for the kangaroo 0.3 package
Artifacts        : kangaroo-0:0.3-1.noarch

Hint: [d]efault, [e]nabled, [x]disabled, [i]nstalled, [a]ctive]
[root@kvm-02-guest19 ~]# dnf module install kangaroo:0:0.2-1.noarch
Updating Subscription Management repositories.
swid_zoo_repo                                                                                            51 kB/s | 3.3 kB     00:00    
Error: Problems in request:
missing groups or modules: kangaroo:0:0.2-1.noarch
[root@kvm-02-guest19 ~]# dnf module install kangaroo:0.2-1.noarch
Updating Subscription Management repositories.
swid_zoo_repo                                                                                            52 kB/s | 3.3 kB     00:00    
Error: Problems in request:
missing groups or modules: kangaroo:0.2-1.noarch
 
```

Reply:
can't we use `0.2-1` and `0.3-1`?

Reply:
```
[root@kvm-02-guest19 ~]# dnf module install kangaroo:0.2-1
Updating Subscription Management repositories.
swid_zoo_repo                                                                                            52 kB/s | 3.3 kB     00:00    
Error: Problems in request:
missing groups or modules: kangaroo:0.2-1
[root@kvm-02-guest19 ~]# dnf module install kangaroo:0:0.2-1
Updating Subscription Management repositories.
swid_zoo_repo                                                                                           647  B/s | 3.3 kB     00:05    
Error: Problems in request:
missing groups or modules: kangaroo:0:0.2-1
[root@kvm-02-guest19 ~]# 


```

Reply:
ok, so we are working with repositories that will be always static? Like in 6 months, 1 year, 2 years? I mean that number will never change?

Reply:
This is a static repository.

Reply:
I still don't like that number, so if second reviewer is fine and if you add comment explaining where you take that number, then okay, let's move forward.

Reply:
@oshtaier If you have used RHEL8 repos which also show the version similar to this. This is one of the outputs from RHEL8 beaker repo. I hope this ouput satisfies you.

```
[root@<hostname> ~]# dnf module info nodejs
Updating Subscription Management repositories.
swid_zoo_repo                                                                                                                                 15 kB/s | 3.3 kB     00:00    
Name             : nodejs
Stream           : 10 [d][a]
Version          : 820190108092226
Context          : 9edba152
Profiles         : common [d], development, minimal, s2i
Default profiles : common
Repo             : beaker-AppStream
Summary          : Javascript runtime
Description      : Node.js is a platform built on Chrome's JavaScript runtime for easily building fast, scalable network applications. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient, perfect for data-intensive real-time applications that run across distributed devices.
Artifacts        : nodejs-1:10.14.1-1.module+el8+2632+6c5111ed.x86_64
                 : nodejs-devel-1:10.14.1-1.module+el8+2632+6c5111ed.x86_64
                 : nodejs-docs-1:10.14.1-1.module+el8+2632+6c5111ed.noarch
                 : nodejs-nodemon-0:1.18.3-1.module+el8+2632+6c5111ed.noarch
                 : nodejs-packaging-0:17-2.module+el8+2632+6c5111ed.noarch
                 : npm-1:6.4.1-1.10.14.1.1.module+el8+2632+6c5111ed.x86_64
 

```

Reply:
@omkarkhatavkar oh, I did not mean anything related to application. With that, I completely trust you. I just don't like such things to be present in test automation